### PR TITLE
fix(clerk-js): Update `createFlexGapPropertyIosCompat` user agent regex check

### PR DESCRIPTION
**Pull request by alexcarpenter**
## Description

Fixes issue where `createFlexGapPropertyIosCompat` was incorrectly detecting a chrome browser agent.

Chrome user agent example:

`"Mozilla/5.0 (iPhone; CPU iPhone OS 17_5 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/128.0.6613.92 Mobile/15E148 Safari/604.1"`

Our regex was matching the `13` here `128.0.6613.92`

### iOS Chrome:

| BEFORE | AFTER |
|--------|--------|
| IMG_2198 | IMG_2199 |

Fixes SDKI-626

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

**Comment by alexcarpenter** [comment]
!preview

**Comment by clerk-cookie** [comment]
Hey @alexcarpenter, your preview is currently deploying and should be available shortly.

| Status | Preview | Updated (UTC) |
| :-- | :-- | :-- |
| :cook: _Deploying..._ | [Inspect](https://javascript-preview-9hi1usy0e.clerkstage.dev) | Aug 27, 2024 05:40 PM |

**Comment by alexcarpenter** [comment]
!preview

File: packages/clerk-js/src/ui/primitives/gapPropertyCompat.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
-export const createFlexGapPropertyIosCompat = (val: string) => {
+export const createFlexGapPropertyIosCompat = (val: string, dir: 'col' | 'row') => {
   // Resolve gap property compatibility issue on iOS 13
   // This is not needed for css grid, only for flex
-  // TODO: Test whether we need to make the check more generic
-  if (navigator?.userAgent?.match(/(iphone|ipad).+os.+13.+safari/i)) {
+  if (navigator?.userAgent?.match(/(iphone|ipad).+(os).*(\s13).+safari/i)) {
```
Comment (alexcarpenter) [inline packages/clerk-js/src/ui/primitives/gapPropertyCompat.ts:4]: This was the main fix here:

```diff
- navigator?.userAgent?.match(/(iphone|ipad).+os.+13.+safari/i)
+ navigator?.userAgent?.match(/(iphone|ipad).+(os).*(\s13).+safari/i)
```

File: packages/clerk-js/src/ui/primitives/gapPropertyCompat.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
     return {
-      '& > * + *': {
-        marginTop: val,
+      '& > *:not([hidden]):not([style*="visibility: hidden"]) + *:not([hidden]):not([style*="visibility: hidden"])': {
+        marginLeft: dir === 'row' ? val : undefined,
+        marginTop: dir === 'col' ? val : undefined,
       },
```
Comment (alexcarpenter) [inline packages/clerk-js/src/ui/primitives/gapPropertyCompat.ts:9]: This update was to improve the layout rendering for the fallback:

| BEFORE | AFTER |
|--------|--------|
| Screenshot 2024-08-27 at 5 10 36 PM | Screenshot 2024-08-27 at 5 11 21 PM |

File: packages/clerk-js/src/ui/primitives/gapPropertyCompat.ts (hunk 1 above)
Comment (BRKalow) [inline packages/clerk-js/src/ui/primitives/gapPropertyCompat.ts:4]:
```suggestion
  if (navigator?.userAgent?.match(/(iphone|ipad).+(os).*(\s13_).+safari/i)) {
```

**Review by BRKalow — APPROVED** [review]

File: .changeset/nine-spies-own.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"@clerk/clerk-js": patch
+---
+
+Fixes issue where `createFlexGapPropertyIosCompat` was incorrectly detecting a chrome browser agent
```
Comment (BRKalow) [inline .changeset/nine-spies-own.md:5]:
```suggestion
Fixes issue where `createFlexGapPropertyIosCompat` was incorrectly detecting a modern Chrome for iOS user agent.
```

**Pull request merged**